friends/friendsController.js

```diff
@@ -0,0 +1,72 @@
+const router = require('express').Router();
+
+const Friend = require('./friendsModel');
+
+router
+    .route('/')
+    .get((req, res) => {
+        Friend
+            .find()
+            .then(friends => {
+                if (req.age < 1 || req.age > 120) {
+                    res.status(400).json({ errorMessage: "Please provide firstName, lastName and age for the friend." });
+                }  
+                res.status(200).json(friends);
+            })
+            .catch(err => {
+                res.status(400).json({ errorMessage: "Please provide firstName, lastName and age for the friend." });
+            });
+    })
+
+    .post((req, res) => {
+        const { firstName, lastName, age } = req.body;
+        const newFriend = new Friend({ firstName, lastName, age });
+        newFriend
+            .save()
+            .then(savedFriend => {
+                res.status(201).json(savedFriend);
+            })
+            .catch(err => {
+                res.status(422).json({ err: 'Error' });
```

Comment:
I don't know if you caught this in the CS10 channel, but Dre'Sean shared a cool trick: if you use ({ error: error.message }) you get a more descriptive error message in your json response (or console.log statement).